### PR TITLE
Feature/event tracing tests

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
Tests for event tracing

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -567,7 +567,7 @@ void USpatialReceiver::OnAuthorityChange(const Worker_AuthorityChangeOp& Op)
 
 void USpatialReceiver::HandlePlayerLifecycleAuthority(const Worker_AuthorityChangeOp& Op, APlayerController* PlayerController)
 {
-	UE_LOG(LogSpatialReceiver, Verbose, TEXT("HandlePlayerLifecycleAuthority for PlayerController %s."),
+	UE_LOG(LogSpatialReceiver, Verbose, TEXT("HandlePlayerLifecycleAuthority for PlayerController %d."),
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp:570]: Hmm is this correct?

**Review by improbable-danny — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp:570]: not sure where this came from 🤔

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp [hunk 2]
```diff
@@ -227,4 +227,14 @@ Trace_SpanId SpatialEventTracer::GetSpanId(const EntityComponentId& Id) const
 
 		return *SpanId;
 }
+FString SpatialEventTracer::SpanIdToString(const Trace_SpanId& SpanId)
+{
+	FString HexStr;
+	for (int i = 0; i < 16; i++)
+	{
+		HexStr += FString::Printf(TEXT("%02x"), SpanId.data[i]);
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:235]: Another option is %02x%02x%02x%02x to avoid cycling strings

Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:235]: Or taking advantage of %02x takes an unsigned integer which should cover at least 4 SpanId elements (needs to be 0-padded for higher widths)

Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:235]: We could just avoid this in critical paths though

**Review by mironec — APPROVED** [review]
Though there was that random change, I think the rest is the same as I reviewed last time.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
Comment (yujunting) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp:570]: https://github.com/spatialos/UnrealGDK/pull/2622

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp (hunk 2 above)
Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:235]: Sorry, what is the suggested change here?

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/ComponentUpdateEventTracingTest.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+void AComponentUpdateEventTracingTest::FinishEventTraceTest()
+{
+	int EventsTested = 0;
+	int EventsFailed = 0;
+	for (const auto& Pair : TraceEvents)
+	{
+		const FString& SpanIdString = Pair.Key;
+		const FName EventName = Pair.Value;
```
Comment (yujunting) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/ComponentUpdateEventTracingTest.cpp:21]: Why not const FName& EnventName?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp [hunk 4]
```diff
@@ -210,7 +210,7 @@ void SpatialEventTracer::UpdateComponent(const Worker_Op& Op)
 
 	Trace_SpanId MergeCauses[2] = { Op.span_id, OldSpanId };
 	TOptional<Trace_SpanId> SpanId = CreateSpan(MergeCauses, 2);
-	TraceEvent(FSpatialTraceEventBuilder::CreateMergeComponent(Id.EntityId, Id.ComponentId), SpanId);
+	TraceEvent(FSpatialTraceEventBuilder::CreateMergeComponentUpdate(Id.EntityId, Id.ComponentId), SpanId);
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:213]: Are these changes meant to be in this PR?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp (hunk 2 above)
Comment (improbable-danny) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:235]:
```suggestion
		TCHAR Buffer[3];
		if (TCString<TCHAR>::Snprintf(Buffer, sizeof(Buffer), TEXT("%02X"), SpanId.data[i]) > 0)
		{
			HexStr += Buffer;
		}
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp (hunk 4 above)
Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:213]: Yes and no... The reason I changes this into two events was to make it easier to test.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp (hunk 2 above)
Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp:235]: Will make this change in the user event tracing PR.

**Pull request merged**